src/main/java/edu/ie3/datamodel/io/factory/timeseries/CosmoTimeBasedWeatherValueFactory.java

```diff
@@ -72,14 +81,24 @@ protected TimeBasedValue<WeatherValue> buildModel(TimeBasedWeatherValueData data
         data.getQuantity(WIND_DIRECTION, StandardUnits.WIND_DIRECTION);
     ComparableQuantity<Speed> windVelocity =
         data.getQuantity(WIND_VELOCITY, StandardUnits.WIND_VELOCITY);
+
+    ComparableQuantity<Temperature> groundTemperature = null;
+    try {
+      groundTemperature = data.getQuantity(GROUND_TEMPERATURE, StandardUnits.TEMPERATURE);
+    } catch (IllegalArgumentException ignored) {
+
+    }
```

Comment:
In my opinion this is a bit counterintuitive. The advantage of try is that one get's a result of the try. But we ignore this. 

But first, please change to `Optional<ComparableQuantity<Tempearture>>` . And then it should be possible to try to get the quantity from data, as for the others, and wrap it into some Optional.ofNullable, or similar.